js/localStorage.js

```diff
@@ -0,0 +1,26 @@
+const name1 = document.querySelector('name');
+const email1 = document.querySelector('email');
+const message1 = document.querySelector('message');
+
+function storeValues(name1, email1, message1) {
+  const dataObject = JSON.stringify({ name1, email1, message1 });
+  localStorage.setItem('dataObject', dataObject);
+}
+
+function populateStorage() {
+  const formValues = JSON.parse(localStorage.getItem('dataObject'));
+  if (formValues) {
+    name1.value = formValues.name1;
+    email1.value = formValues.email1;
+    message1.value = formValues.message1;
+  }
+}
+
+function getValues() {
+  name1.addEventListener('input', () => storeValues(name1.value, email1.value, message1.value));
+  email1.addEventListener('input', () => storeValues(name1.value, email1.value, message1.value));
+  message1.addEventListener('input', () => storeValues(name1.value, email1.value, message1.value));
+}
```

Comment:
- [ ] The data is not being preserved upon refreshing the page, kindly fix it. You could try using an on-load function with an if statement inside that checks if there's any saved data in the local storage and if so the input fields are pre-filled with the data.
For example
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
````
`